middlewares/logger_test.go

```diff
@@ -41,14 +39,21 @@ var (
 )
 
 func TestLogger(t *testing.T) {
-	if runtime.GOOS == "windows" {
-		logfilePath = filepath.Join(os.Getenv("TEMP"), logfileName)
-	} else {
-		logfilePath = filepath.Join("/tmp", logfileName)
+	tmp, err := ioutil.TempDir("", "testlogger")
+	if err != nil {
+		t.Fatal("failed to create temp dir: ", err)
 	}
+	defer os.RemoveAll(tmp)
+
+	logfilePath := filepath.Join(tmp, logfileNameSuffix)
 
 	logger = NewLogger(logfilePath)
-	defer cleanup()
+	defer logger.Close()
+
+	if _, err := os.Stat(logfilePath); err != nil {
```

Comment:
I think you need [os.IsNotExist](http://stackoverflow.com/a/12518877/420061), don't you?